src/manage_nsfs/manage_nsfs_glacier.js

```diff
@@ -36,13 +36,38 @@ async function process_migrations() {
  */
 async function run_glacier_migrations(fs_context, backend) {
     // This WAL is getting opened only so that we can process all the prcess WAL entries
-    const wal = new PersistentLogger(
+    const wal = new PersistentLogger(config.NSFS_GLACIER_LOGS_DIR, GlacierBackend.MIGRATE_WAL_NAME, { locking: 'EXCLUSIVE' });
+    const failure_wal = new PersistentLogger(
         config.NSFS_GLACIER_LOGS_DIR,
-        GlacierBackend.MIGRATE_WAL_NAME,
-        { disable_rotate: true, locking: 'EXCLUSIVE' },
+        `${GlacierBackend.MIGRATE_WAL_NAME}.failure`,
+        { locking: 'EXCLUSIVE' },
```

Comment:
I think the term wal here is unrelated anymore. I would just call it log and failure_log perhaps